apm-sniffer/apm-sdk-plugin/spring-plugins/mvc-annotation-commons/src/main/java/org/apache/skywalking/apm/plugin/spring/mvc/commons/interceptor/AbstractMethodInterceptor.java

```diff
@@ -208,4 +221,31 @@ private void collectHttpParam(HttpServletRequest request, AbstractSpan span) {
             Tags.HTTP.PARAMS.set(span, tagValue);
         }
     }
+
+    private boolean shouldCollectHeader() {
+        List<String> includeHeaders = SpringMVCPluginConfig.Plugin.Http.INCLUDE_HTTP_HEADERS;
+        return includeHeaders != null && includeHeaders.size() > 0 ;
+    }
+
+    private void collectHttpHeaders(HttpServletRequest request, AbstractSpan span) {
+        final Enumeration<String> headerNames =  request.getHeaderNames();
+        if (headerNames == null) {
+            return;
+        }
+        final Map<String, String[]> headersMap = new HashMap<>();
```

Comment:
Please use the Array list rather than Map. The map is a heavy data structure, it costs more CPU, memory, and has more chances to trigger GC

Reply:
done